### ISSUE TITLE
FR - release for Android x86

### ISSUE THREAD

**Issue by kvg84594** [bettercap/bettercap#124]
Hello,

It would be great if there is a precompiled release for Android x86 (e.g. Intel Atom android devices), not only for ARM.

**Comment by alexmozzhakov** [comment]
Did u try [this](https://github.com/bettercap/bettercap/wiki/Cross-Compilation-(-ARM-example-))

**Comment by evilsocket** [comment]
@alexmozzhakov for Android the process is different and I can understand users requesting for precompiled binaries, I have a hard time compiling it myself :)

@kvg84594 will add the build in a future release.

**Comment by fat-tire** [comment]
FWIW I built it on a Samsung Chromebook Pro (intel m3) in Termux with the standard `go get` command resulting in:
```
$ ./bettercap
bettercap v2.2 (type 'help' for a list of commands)

socket for SIOCETHTOOL(ETHTOOL_GET_TS_INFO): Permission denied
```

Leaving aside the fact I haven't rooted the machine which I'm pretty sure accounts for the permission denied, it built fine without any errors, created the binary and `bettercap --help` works...

**Comment by yungtravla** [comment]
@fat-tire you need a rooted device

**Comment by fat-tire** [comment]
I know.  I am only saying that you can build on x86 Android devices w/Termux using a chromebook, since that's how the arm version is built in [build.sh](https://github.com/bettercap/bettercap/blob/master/build.sh#L171).  I tested it and it built.  Just added the caveat that I hadn't rooted the machine so didn't test its functionality.

**Comment by yungtravla** [comment]
@fat-tire the `go build` function does exactly that, it builds it. It does not run it.

What you could do is:

1. root your device
2. install/build golang and bettercap
3. research the errors, fix them and report it here

That we can work with, but not with a `permission denied` error :)

**Comment by tomford007** [comment]
I downloaded precompiled arm version to my android and can't figure out install. Tried about an hour. Its not an apk.

**Comment by yungtravla** [comment]
@tomford007 execute the binary in a terminal with `sudo`

**Comment by fat-tire** [comment]
Or to build it yourself (arm or x86) and run in a more complete Linux environment than what you get in the plain Android shell:

1.  From the play store or f-droid, install [Termux](https://termux.com/).
2.  In Termux, type:  `pkg install golang git libpcap-dev` and say "y" to the dependencies.
3.  Type `go get github.com/bettercap/bettercap` and wait...
4.  Your build is in `go/bin/bettercap`. 

I don't think it's statically built, so you may need to use something like [this](https://github.com/st42/termux-sudo) to run it in Termux.

I haven't tested the build on a rooted device but maybe someone else can give it a shot and report back.

**Comment by evilsocket** [comment]
Yep the procedure that @fat-tire described is precisely what I used in order to have an Android build env on my NVIDIA Shield tablet, that's the ideal config to compile and run bettercap on Android.

**Comment by whc200187** [comment]
I tried ur way @fat-tire and it really works! But I think the wireless adapter of my phone is too weak to handle this tool.
![screenshot_20180406-174238](https://user-images.githubusercontent.com/11062281/38414927-793b289c-39c2-11e8-81e6-51ab0ff74af8.png)

**Comment by ScriptTiger** [comment]
Just to circle back to the OP's, @kvg84594's, original question about Android x86, I would say if you're looking for a minimal operating system to run bettercap on a PC, I don't think you can beat Tiny Core. Android has a lot of hype, but it's only useful if you really like the point-and-click look and feel that the mobile OS was designed for.or you really like Android games. However, trying to use Android x86 as a minimal OS or for penetration testing in general is probably not the best platform choice, especially when bettercap is inherently a text-based utility, apart from the web GUI.

You can download the prebuilt Tiny Core packages here to test them out:
https://tinycorefactory.github.io/penetration-testing/

The open-source build script for the above packages is located here:
https://github.com/TinyCoreFactory/Assembly-Line/blob/master/Penetration-Testing/betterbox.sh
If you find any issues with the packages, you can report them there.

The above packages work the same as advertised by bettercap's usage documentation if you're looking to get started:
https://www.bettercap.org/

Since Tiny Core is designed to work with embedded systems, similar to Android, the UI and caplets are already included and cannot be updated automatically by bettercap since it's impossible to know whether you will have external storage available or not and what those devices might be, if any. However, you can choose to manually update your UI and caplets and make a custom Tiny Core restore file to be saved to an external storage device that can be loaded automatically at boot to overwrite the static UI and caplets that come with the packages. You will also need to do this if you want to keep any changes you make to your caplets after reboot, such as credentials, custom port numbers, etc. If you are familiar with embedded systems, this process is similar to loading a boot file at start-up. If there is no restore file available, Tiny Core will boot a pristine image every time, with everything contained within a single ram disk that disappears after shutdown and all changes lost.

For more documentation on Tiny Core, I would recommend checking out their official website:
https://distro.ibiblio.org/tinycorelinux

**Comment by evilsocket** [comment]
@ScriptTiger the "open-source build script" is not a build script at all as it only downloads the precompiled binaries and installs them.

**Comment by ScriptTiger** [comment]
Yes, true. I wasn't sure how else to word that. It doesn't compile anything from source to binaries, but it builds the squash structures with the needed dependencies and links to work with Tiny Core, and uses various compression techniques to optimize it for embedded. I don't think there is a need to compile it as there are already releases made available that are compatible with Tiny Core and development builds are not reliable and must be built more frequently. @evilsocket, I can add in the necessary steps to build it from source if you would like, but I would just make it an optional step as that would be beyond most general users' needs. Thank you for replying though, any other feedback would be most appreciated.

And as a note to you personally, I take it you have read the script, so I'll just highlight that I also included your documentation with the distro, licensing, readme, etc., but if you would like any other changes or anything at all, just let me know. Thank you, again!

**Comment by spitfire55** [comment]
I got a cross-compilation solution working via Docker for ARM-based Android devices. Here's the repo: https://github.com/spitfire55/gopacket_android
